lib/active_model/serializer/adapter/json.rb

```diff
@@ -37,10 +39,35 @@ def serializable_hash(options = nil)
                   end
               end
             end
-            @result = @core.merge @hash
+
+            if sideload
+              association_ids = {}
+              @hash.map do |association_name, associated_models|
+                # TODO: use active support inflectors?
```

Comment:
I'd say yes, but I'm concerned with ppl using AMS outside Rails context.
I'd keep it like that and come back at it after Rails 5 release